File: 1768-merge-strings-alternately/1768-merge-strings-alternately.java
class Solution {
    public String mergeAlternately(String w1, String w2) {
        int n = w1.length(), m = w2.length(), i = 0, j = 0;
        StringBuilder res = new StringBuilder();
        while (i < n || j < m) {
            if (i < w1.length())
                res.append(w1.charAt(i++));
            if (j < w2.length())
                res.append(w2.charAt(j++));
        }
        return res.toString();
    }
}

File: 1768-merge-strings-alternately/1768-merge-strings-alternately.cpp
class Solution {
public:
    string mergeAlternately(string word1, string word2) {
        int n = word1.length(), m = word2.length(), i = 0, j = 0;
        string res = "";
        while(i < n || j < m){
            if(i < n) res += word1[i++];
            if(j < m) res += word2[j++];
        }
        return res;
    }
};